lib/Basic/Version.cpp

```diff
@@ -317,28 +317,25 @@ Optional<Version> Version::getEffectiveLanguageVersion() const {
     return None;
   }
 
-  // FIXME: When we switch to Swift 5 by default, the "4" case should return
-  // a version newer than any released 4.x compiler, and the
-  // "5" case should start returning getCurrentLanguageVersion. We should
-  // also check for the presence of SWIFT_VERSION_PATCHLEVEL, and if that's
-  // set apply it to the "3" case, so that Swift 4.0.1 will automatically
-  // have a compatibility mode of 3.2.1.
-  switch (Components[0]) {
+   // FIXME: When switching to a new major version by default---say, Swift 4---
+   // the "3" case should return a version newer than any released 3.x compiler,
+   // and the "4" case should start returning getCurrentLanguageVersion.  switch (Components[0]) {
   case 3:
 #ifdef SWIFT_VERSION_PATCHLEVEL
     return Version{3, 4, SWIFT_VERSION_PATCHLEVEL};
 #else
     return Version{3, 4};
 #endif
   case 4:
-    static_assert(SWIFT_VERSION_MAJOR == 4,
-                  "getCurrentLanguageVersion is no longer correct here");
     // Version '4' on its own implies '4.1.50'.
     if (size() == 1)
       return Version{4, 1, 50};
     return Version::getCurrentLanguageVersion();
```

Comment:
This line here's no longer going to be right. Maybe the whole section should be "If Version >= 4.2, return 4.2, else return 4.1.50".